lib/cog/models/ecto_json.ex

```diff
@@ -52,6 +52,37 @@ defmodule Cog.Models.EctoJson do
     end
   end
 
+  def render_permissions(model, opts \\ []) do
+    opts = Keyword.merge(@render_defaults, opts)
+
+    policy = Keyword.fetch!(opts, :policy)
+    data = do_render(model, policy)
+    |> format_permissions(model)
+
+    case Keyword.fetch!(opts, :envelope) do
+      nil ->
+        data
+      envelope ->
+        Map.put(%{}, envelope, data)
+    end
+  end
+
+  def format_permissions(data, []), do: data
+  def format_permissions(data, %{__struct__: Cog.Models.Role}), do: data
+  def format_permissions(data, model) when is_list(model) do
+    [item | _] = model
+    format_permissions(data, item)
+  end
+  def format_permissions(data, _model) when is_list(data) do
+    perms = Enum.group_by(data, &(Map.get(&1, :namespace)))
```

Comment:
You can drop the parens around `&(Map.get(&1, :namespace))` when you're calling a function like this, but it's mostly a style thing.